ios/browser/api/ads/brave_ads.mm

```diff
@@ -252,52 +244,49 @@ - (void)initServiceWithSysInfo:(BraveAdsSysInfo*)sysInfo
     return completion(/*success=*/false);
   }
 
-  [self initDatabase];
-
-  ads = brave_ads::Ads::CreateInstance(*adsClient);
-
   auto cppSysInfo =
       sysInfo ? sysInfo.cppObjPtr : brave_ads::mojom::SysInfo::New();
-  ads->SetSysInfo(std::move(cppSysInfo));
   auto cppBuildChannelInfo = buildChannelInfo
                                  ? buildChannelInfo.cppObjPtr
                                  : brave_ads::mojom::BuildChannelInfo::New();
-  ads->SetBuildChannel(std::move(cppBuildChannelInfo));
-  ads->SetFlags(brave_ads::BuildFlags());
-
   auto cppWalletInfo =
       walletInfo ? walletInfo.cppObjPtr : brave_ads::mojom::WalletInfoPtr();
-  ads->Initialize(std::move(cppWalletInfo),
-                  base::BindOnce(^(const bool success) {
-                    [self periodicallyCheckForAdsResourceUpdates];
-                    [self registerAdsResources];
-                    if (success) {
-                      [self notifyDidInitializeAds];
-                    }
-                    completion(success);
-                    if (!success) {
-                      [self cleanupAds];
-                    }
-                  }));
+
+  ProfileIOS* profile = [self getLastUsedProfile];
+  adsService = brave_ads::AdsServiceFactoryIOS::GetForBrowserState(profile);
```

Comment:
Just fyi, this will always be the regular profile (not private/OTR), which matches the current behaviour anyways, but just figured I'd point it out

Reply:
@aseren could we please add a comment explaining this? Thanks

Reply:
Or maybe we add a CHECK instead of a comment?

Reply:
There's actually a `CHECK` under this line already :)

Reply:
Added an explanatory comment